packages/core/src/models/promptVariants.ts

```diff
@@ -46,9 +46,16 @@ export class PromptVariants {
 
     const { name, appId: appSlug, inputSchema, outputSchema } = this.definition;
 
-    const app = await this.prisma.app.findFirstOrThrow({
-      where: { slug: appSlug },
-    });
+    let app: App | undefined = undefined;
+
+    try {
+      app = await this.prisma.app.findFirstOrThrow({
```

Comment:
I have an opinion about this particular pattern, but it's the norm for the project so it doesn't reflect on this PR. I'll share on slack as it's part of a wider conversation